From 76f9b514c0ec41c6ff43ffa378687ce0e67a420c Mon Sep 17 00:00:00 2001
Message-Id: <76f9b514c0ec41c6ff43ffa378687ce0e67a420c.1424394676.git.feitong.yi@intel.com>
In-Reply-To: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
References: <00f4430a265007e8e2c016cc7926bf530835ea30.1424394676.git.feitong.yi@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Wed, 25 Jun 2014 14:43:34 +0100
Subject: [PATCH 39/61] REVERTME [VPG]: drm/i915: Use OOM adjust value as
 scheduler priority

The Android Out Of Memory adjustment value is a good approximation for GPU
workload priority. It is set to zero for all regular applications and non-zero
for important system tasks. Thus, it can be used to say that SurfaceFlinger is
high priority and everything else is low.

The final intention is to replace this patch with code to explicitly set the
priory from SurfaceFlinger using a dedicated IOCTL to the driver. When that has
been implemented, this patch can be dropped.

Change-Id: I67e792cca00adb588e3d0230e55898b83f615d5c
For: VIZ-1587
For: VIZ-4741
For: GMIN-3638
Signed-off-by: John Harrison <John.C.Harrison@Intel.com>
---
 drivers/gpu/drm/i915/i915_gem_execbuffer.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem_execbuffer.c b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
index 198b733..13dba13 100644
--- a/drivers/gpu/drm/i915/i915_gem_execbuffer.c
+++ b/drivers/gpu/drm/i915/i915_gem_execbuffer.c
@@ -1637,6 +1637,12 @@ i915_gem_do_execbuffer(struct drm_device *dev, void *data,
 	params->args_DR4                = args->DR4;
 	params->batch_obj               = batch_obj;
 
+	/* Use the out-of-memory priority value as a suitable starting point for
+	 * the buffer priority. It seems to be zero for application level tasks
+	 * and less than zero for system tasks. */
+	qe.priority = (current->signal->oom_score_adj < 0) ?
+					-current->signal->oom_score_adj : 0;
+
 	/*
 	 * Save away the list of objects used by this batch buffer for the
 	 * purpose of tracking inter-buffer dependencies.
-- 
1.7.9.5

